src/time.rs

```diff
@@ -0,0 +1,278 @@
+//! Wall clock time
+//!
+//! Time is fixed to nanosecond precision, with user supplied resolution.
+
+/// 61 bit nanosecond wall clock time
+///
+/// Note that the upper 3 bits are reserved for internal use.
+///
+/// This can represent approximately 73 years.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[repr(transparent)]
+pub struct Nanoseconds(u64);
+
+/// Nanosecond time resolution
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[repr(transparent)]
+pub struct NanosecondResolution(pub u32);
+
+/// Nanosecond high bits
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+#[repr(transparent)]
+pub struct NanosecondsHighBits(pub u32);
```

Comment:
If the NanosecondsXBits types are supposed to contain opaque bytes for transport (in a fixed endian-ness), I recommend we represent them as [u8;4] instead of u32.

Reply:
These should probably be accompanied by a few helper converters (constructors and extractors) that are explicit about their conversion input and output formats.